code/modules/mob/living/simple_animal/hostile/retaliate/king_of_goats.dm

```diff
@@ -0,0 +1,328 @@
+			melee_damage_type = BURN
+
+		else if(prob(5)) //earthquake spell
+			visible_message("<B><span class='danger'>\The [src]' eyes begin to glow ominously as dust and debris in the area is kicked up in a light breeze!!</span></B>")
+			stop_automated_movement = TRUE
+			if(do_after(src, 6 SECONDS, src))
+				var/health_holder = getBruteLoss()
+				visible_message("<B><span class='cult'>\The [src] raises its fore-hooves and stomps them into the ground with incredible force!!</span></B>")
+				explosion(get_step(src,pick(GLOB.cardinals)), -1, 2, 2, 3, 6)
+				explosion(get_step(src,pick(GLOB.cardinals)), -1, 1, 4, 4, 6)
+				explosion(get_step(src,pick(GLOB.cardinals)), -1, 3, 4, 3, 6)
+				stop_automated_movement = FALSE
+				spellscast += 2
+				if(!getBruteLoss() > health_holder)
+					adjustBruteLoss(health_holder - getBruteLoss()) //our own magicks cannot harm us
+			else
+				visible_message("<span class='notice'>\The [src] loses concentration and huffs haughtily.</span>")
+				stop_automated_movement = FALSE
+
+		else return
+
+/mob/living/simple_animal/hostile/retaliate/goat/king/phase2/proc/phase3_transition()
+	phase3 = TRUE
+	spellscast = 0
+	maxHealth = 750
+	revive(TRUE)
+	current_song = 'sound/ambience/Visager-Miniboss_Fight.ogg'
+	current_song_length = 1759
+	var/sound/song_played = sound(current_song)
+	for(var/mob/M in rangers)
+		if(!M.client || !(M.client.prefs.toggles & SOUND_INSTRUMENTS))
+			continue
+		M.stop_sound_channel(CHANNEL_JUKEBOX)
+		rangers[M] = world.time + current_song_length
+		M.playsound_local(null, null, 30, channel = CHANNEL_JUKEBOX, S = song_played)
+	stun_chance = 10
+	update_icon()
+	visible_message("<span class='cult'>\The [src]' wounds close with a flash, and when he emerges, he's even larger than before!</span>")
+
+
+/mob/living/simple_animal/hostile/retaliate/goat/king/phase2/update_icon()
+	var/matrix/M = new
+	if(phase3)
+		icon_state = "king_goat3"
+		icon_living = "king_goat3"
+		M.Scale(1.5)
+	else
+		M.Scale(1.25)
+	transform = M
+	pixel_y = 10
+
+/mob/living/simple_animal/hostile/retaliate/goat/king/phase2/Life()
+	. = ..()
+	if(stat != DEAD)
+		var/sound/song_played = sound(current_song)
+
+		for(var/mob/M in range(10, src))
+			if(!M.client || !(M.client.prefs.toggles & SOUND_INSTRUMENTS))
+				continue
+			if(!(M in rangers) || world.time > rangers[M])
+				M.stop_sound_channel(CHANNEL_JUKEBOX)
+				rangers[M] = world.time + current_song_length
+				M.playsound_local(null, null, 30, channel = CHANNEL_JUKEBOX, S = song_played)
+		for(var/mob/L in rangers)
+			if(get_dist(src, L) > 10)
+				rangers -= L
+				if(!L || !L.client)
+					continue
+				L.stop_sound_channel(CHANNEL_JUKEBOX)
+	else
+		for(var/mob/L in rangers)
+			rangers -= L
+			if(!L || !L.client)
+				continue
+			L.stop_sound_channel(CHANNEL_JUKEBOX)
+	if(move_to_delay < 3)
+		move_to_delay += 0.1
+	if((health <= 150 && !phase3 && spellscast == 5) || (stat == DEAD && !phase3)) //begin phase 3, reset spell limit and heal
```

Comment:
what if the goatking cast earthquake as its last spell, bringing spellscast up to 6? 6!=5.